### PR TITLE
[DUOS-2027] Fixed signing on always redirecting you to the first available console page

### PR DESCRIPTION
**Pull request by lu-c**
Just that when you are on a page for too long, maybe you were coding in your IDE, your session times out, and the website asks you to log back in. When you log in though, instead of redirecting you back to the page you were on, you find yourself now on "/admin_manage_dar_collections", and you can't even hit the back button to go back to where you were from before the site FORCED you to leave

I know there's definitely a better way to do this (without relying on query parameters in the url bar, and just using the router itself), but i have wasted enough unauthorized time on this unasked for ticket

fast, but dirty. please let this go through XP This just gets annoying during development!

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/utils.js [hunk 1]
```diff
@@ -218,13 +218,20 @@ export const setUserRoleStatuses = (user, Storage) => {
 
 export const Navigation = {
   back: async (user, history) => {
+    const queryParameters = Object.fromEntries(
+      window.location.search
+        .replace('?', '')
+        .split('&')
+        .map(queryString => queryString.split('='))
+    );
```
Comment (connorlbark) [inline src/libs/utils.js]: is there a utility/library function to do this? just for safety purposes

File: src/libs/utils.js [hunk 2]
```diff
@@ -218,13 +218,20 @@ export const setUserRoleStatuses = (user, Storage) => {
...
-            : user.isResearcher ? '/dataset_catalog'
-              : user.isAlumni ? '/summary_votes'
-                : '/';
+      queryParameters.redirectTo ? queryParameters.redirectTo
+        :user.isAdmin ? '/admin_manage_dar_collections'
+          :user.isChairPerson ? '/chair_console'
+            : user.isMember ? '/member_console'
+              : user.isResearcher ? '/dataset_catalog'
+                : user.isAlumni ? '/summary_votes'
+                  : '/';
```
Comment (connorlbark) [inline src/libs/utils.js]:
```suggestion
      queryParameters.redirectTo ? queryParameters.redirectTo
        : user.isAdmin ? '/admin_manage_dar_collections'
          : user.isChairPerson ? '/chair_console'
            : user.isMember ? '/member_console'
              : user.isResearcher ? '/dataset_catalog'
                : user.isAlumni ? '/summary_votes'
                  : '/';
```
nit

File: src/libs/utils.js (hunk 1 above)
Comment (lu-c) [reply src/libs/utils.js]: looks like there's a native browser way to do it 👍

**Review by connorlbark — APPROVED** [review]
👍

**Comment by rushtong** [comment]
I think we have a related ticket to this: https://broadworkbench.atlassian.net/browse/DUOS-1501
Please review to see if there's overlap here.

**Comment by lu-c** [comment]
> I think we have a related ticket to this: https://broadworkbench.atlassian.net/browse/DUOS-1501 Please review to see if there's overlap here.

Oh yeah, this is definitely the same thing! I'm adding it as a url hash because I couldn't figure out the appropriate react-router way to do it... That's perfect! :D

**Review by rushtong — APPROVED** [review]
This works as designed. One thing I'd like to see is the ability to navigate to a pasted link in the address bar by remembering the URL, signing the user in, and then redirecting to the remembered URL. This PR doesn't quite get us to that point and we will definitely need this for TDR/Catalog integration.

**Pull request merged**